app/client/lib/ethereum/observeEvents.js

```diff
@@ -12,15 +14,31 @@ observeEvents = function(){
     */
     collectionObservers[collectionObservers.length] = Events.find({}).observe({
         /**
-        This will observe the transactions creation and create watchers for outgoing transactions, to see when they are mined.
+        This will observe when events are added and link it to the custom contract.
 
         @method added
         */
         added: function(newDocument) {
+            
+            // This creates a temporary cache for the contracts, to reduce the amount of db reads            
+            var customContract = customContractsCache[newDocument.address.toLowerCase()];
+
+            if (typeof customContract == 'undefined') {
+                var customContract = CustomContracts.findOne({address: newDocument.address.toLowerCase()});
```

Comment:
Is it possible to move line 39 into this if statement?
Currently it will always execute this line if the `customContract` isn't in the cache which will only happen after at least 1 unknown event is raised. Until then each time `findOne` is executed.